typing/jkind.ml

```diff
@@ -1529,6 +1600,7 @@ module Builtin = struct
       { layout; mod_bounds = Bounds.max; with_bounds = No_with_bounds }
     in
     fresh_jkind_poly desc ~annotation:None ~why:(Product_creation why)
+    |> mark_best
```

Comment:
This is like the `mark_best` calls in typedecl, and is suspect -- note the `mod_bounds = Bounds.max` which doesn't look very best-like.

Reply:
and in fact we don't need it.